Structure_Engine/Query/Description.cs

```diff
@@ -351,6 +498,202 @@ private static string CheckGetMaterialName(IMaterialFragment material)
+                    return "x⁻";
+                case DOFType.FixedPositive:
+                    return "x⁺";
+                case DOFType.Spring:
+                    return "e";
+                case DOFType.SpringNegative:
+                    return "e⁻";
+                case DOFType.SpringPositive:
+                    return "e⁺";
+                case DOFType.SpringRelative:
+                    return "r";
+                case DOFType.SpringRelativeNegative:
+                    return "r⁻";
+                case DOFType.SpringRelativePositive:
+                    return "r⁺";
+                case DOFType.NonLinear:
+                    return "n";
+                case DOFType.Friction:
+                    return "fᵣ";
+                case DOFType.Damped:
+                    return "d";
+                case DOFType.Gap:
+                    return "g";
+                default:
+                    return "-";
+            }
+        }
+
+        /***************************************************/
+
+        [Description("Returns true if any of the dof elements is of a value based type or if any of the values are non-zero.")]
+        private static bool ElasticValuesWritten(this Constraint3DOF constraint)
+        {
+            if (constraint.UX.ValueBasedDOF())
+                return true;
+
+            if (constraint.UY.ValueBasedDOF())
+                return true;
+
+            if (constraint.Normal.ValueBasedDOF())
+                return true;
+
+            if (constraint.KX != 0)
+                return true;
+
+            if (constraint.KY != 0)
+                return true;
+
+            if (constraint.KNorm != 0)
+                return true;
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [Description("Returns true if any of the dof elements is of a value based type or if any of the values are non-zero.")]
+        private static bool ElasticValuesWritten(this Constraint4DOF constraint)
+        {
+            if (constraint.TranslationX.ValueBasedDOF())
+                return true;
+
+            if (constraint.TranslationY.ValueBasedDOF())
+                return true;
+
+            if (constraint.TranslationZ.ValueBasedDOF())
+                return true;
+
+            if (constraint.RotationX.ValueBasedDOF())
+                return true;
+
+            if (constraint.TranslationalStiffnessX != 0)
+                return true;
+
+            if (constraint.TranslationalStiffnessY != 0)
+                return true;
+
+            if (constraint.TranslationalStiffnessZ != 0)
+                return true;
+
+            if (constraint.RotationalStiffnessX != 0)
+                return true;
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [Description("Returns true if any of the dof elements is of a value based type or if any of the values are non-zero.")]
```

Comment:
Returns true if any of the degrees of freedom is of a value based type or the values are non-zero.